app/views/layouts/_header.html.erb

```diff
@@ -0,0 +1,46 @@
+<div class="header-partial">
+  <nav class="navbar bg-light">
+    <div class="container-fluid">
+      <div class="d-flex">
+        <h1 class="navbar-brand">
+          <%=link_to "GateExchange", courses_path%>
+        </h1>
+        <div class="navbar-nav me-auto mb-2 mb-lg-0">
+          <div class="d-flex">
+            <% if @course.nil? %>
+              <%  if current_page?(posts_path) %>
```

Comment:
Yeah, this looks good. I am wondering if it is preferable over JS. Just wondering your opinion for personal knowledge.

Reply:
I'd say it's about the same but if we can implement something serverside, and doing it on the frontend doesn't provide any additional advantage, I'd lean with the serverside implementation

Reply:
ok ok sounds good. Yea I think for latency purposes, doing it on the frontend is probably the better way to do it. Especially if we open this to the public and it becomes scalable. It would also be less costly too 🤔